### PR TITLE
Feature/add option to return object

### PR DESCRIPTION
**Pull request by daveobriencouk**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
-  "name": "bytes",
+  "name": "@daveobriencouk/bytes",
   "description": "Utility to parse a string bytes to bytes and vice-versa",
-  "version": "3.0.0",
+  "version": "3.1.0",
```
Comment (dougwilson) [inline package.json]: Please don't bump the version; I can bump during the release.

File: package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
-  "name": "bytes",
+  "name": "@daveobriencouk/bytes",
```
Comment (dougwilson) [inline package.json]: I think you changed the name here accidentally; I can't accept the PR if it'll rename the module :) Just revert this from the PR 👍

File: Readme.md [hunk 3]
```diff
@@ -69,6 +70,8 @@ bytes(1024 * 1.7, {decimalPlaces: 0});
 bytes(1024, {unitSeparator: ' '});
 // output: '1 KB'
 
+bytes(1024, {outputAsObject: true});
+// output: {value: '1',unit: 'KB'}
```
Comment (dougwilson) [inline Readme.md]: I think Node would output a space after the `,` right?

File: Readme.md [hunk 4]
```diff
@@ -44,6 +44,7 @@ Format the given value in bytes into a string. If the value is negative, it is k
 | thousandsSeparator | `string`｜`null` | Example of values: `' '`, `','` and `.`... Default value to `''`. |
 | unit | `string`｜`null` | The unit in which the result will be returned (B/KB/MB/GB/TB). Default value to `''` (which means auto detect). |
 | unitSeparator | `string`｜`null` | Separator to use between number and unit. Default value to `''`. |
+| outputAsObject | `boolean`｜`null` | Whether to return a string, or an object that returns a unitless value and the unit. Default value to `false` |
```
Comment (dougwilson) [inline Readme.md]: I see the idea you're trying to achieve but I don't like the interface you used for it -- varying the output type based on an option can lead to surprising things. Why not simply make this as a new method altogether? 'm not sure. The title of this section and the corresponding jsdoc no longer have the correct return type.

In reality, it seems like you could simply split the output on the `unitSeparator` anyway, like `bytes.format(24211, { unitSeparator: '|' }).split('|')`

File: Readme.md (hunk 3 above)
Comment (captDaylight) [reply Readme.md]: I'm seeing spaces around brackets and between values `{ value: '1', unit: 'KB' }` when I output to node.

**Comment by dougwilson** [comment]
I'm going to close this for now since it's been so long since I left a review and there hasn't been any follow up work or comments. Feel free to re-open when you have time to make adjustments!

**Pull request closed without merging**